aioredlock/algorithm.py

```diff
@@ -67,6 +71,20 @@ def _validate_retry_delay(self, attribute, value):
     def log(self):
         return logging.getLogger(__name__)
 
+    async def __auto_extend(self, lock, count):
+        """
+        Tries to reset the lock's lifetime to lock_timeout every 0.6*lock_timeout automatically
+        In case of fault the LockError exception will be raised
+        :param lock: :class:`aioredlock.Lock`
+        :raises: LockError in case of fault
+        """
+        if count:
```

Comment:
If `extend` will accept `auto_extend_count` kwarg watchdog creation might be moved to `extend` itself and replaced by `await self.extend(lock, count-1)` here.